### PR TITLE
KOGITO-3641 Deploy after set next snapshot

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-3641

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by radtriste** [comment]
Tested ok on a fake release release pipeline

File: Jenkinsfile.promote [hunk 1]
```diff
@@ -246,6 +258,7 @@ void checkoutRepo(String repo) {
     checkout(githubscm.resolveRepository(repo, getGitAuthor(), getBuildBranch(), false))
     // need to manually checkout branch since on a detached branch after checkout command
     sh "git checkout ${getBuildBranch()}"
+    setupMavenConfig()
```
Comment (rsynek) [inline Jenkinsfile.promote:261]: Nitpicking: `setupMavenConfig()` is not what I expect to happen if I call `checkoutRepo`.

**Review by rsynek — APPROVED** [review]
1 optional suggestion, otherwise LGTM.

File: Jenkinsfile.promote (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile.promote:261]: true...gonna update that

**Review by evacchi — APPROVED** [review]

**Pull request merged**